wm/desk.go

```diff
@@ -292,14 +291,66 @@ func (x *x11WM) runLoop() {
 				x.altTabList = nil
 				xproto.UngrabKeyboard(x.x.Conn(), xproto.TimeCurrentTime)
 			}
+		case randr.ScreenChangeNotifyEvent:
+			if x.screenChangeTimestamp == ev.Timestamp {
+				break
+			}
+			x.screenChangeTimestamp = ev.Timestamp
+			desk := desktop.Instance()
+			if desk == nil {
+				break
+			}
+			desk.Screens().RefreshScreens()
+			x.layoutRoots()
 		}
 	}
 
 	fyne.LogError("X11 connection terminated!", nil)
 }
 
+func (x *x11WM) isRootTitle(title string) bool {
+	return strings.Index(title, ui.RootWindowName) == 0
+}
+
+func screenNameFromRootTitle(title string) string {
+	if len(title) <= len(ui.RootWindowName) {
+		return ""
+	}
+	return title[len(ui.RootWindowName):]
+}
+
+func (x *x11WM) getWindowFromScreenName(screenName string) xproto.Window {
+	for _, id := range x.rootIDs {
+		name := windowName(x.x, id)
+		if !x.isRootTitle(name) {
+			continue
+		}
+		if screenNameFromRootTitle(name) == screenName {
+			return id
+		}
+	}
+	return 0
+}
+
+func (x *x11WM) layoutRoots() {
+	for _, screen := range desktop.Instance().Screens().Screens() {
+		win := x.getWindowFromScreenName(screen.Name)
+		if win != 0 {
```

Comment:
I think these is one of those 0 win is error situations that means it would me a little more readable with if win == 0 { continue } - so the main body of the loop is one level higher.